packages/default-reporter/src/reporterForClient/reportProgress.ts

```diff
@@ -156,16 +171,34 @@ function getProgessStatsPushStreamByRequirer (progress$: most.Stream<ProgressLog
   return progessStatsPushStreamByRequirer
 }
 
+function getImportingMethodPushStream (importing$: most.Stream<ImportingLog>) {
+  const importingStatsPushStream = new PushStream<string>()
+
+  let previousImportingMethod = 'hard linked'
+  importing$
+    .forEach((log: ImportingLog) => {
```

Comment:
it is a bad idea to listen to the importing events. These events will happen lots of times. Once for each package. And we really need a single event that tells us whether the copying or hard linking was used.

I guess the most optimal way to do this would be creating a new logger that will only log a single message saying what importing method was used. So it would need to run in package store, in this file: https://github.com/pnpm/pnpm/blob/master/packages/package-store/src/storeController/createImportPackage.ts

Something like:

```ts
function createAutoImporter () {
  let auto = initialAuto

  return auto

  async function initialAuto (
    to: string,
    opts: {
      filesMap: Record<string, string>,
      force: boolean,
      fromStore: boolean,
    }
  ) {
    try {
      await clonePkg(to, opts)
      packageImportMethodLogger.debug({ method: 'clone' }) // ADDED
      auto = clonePkg
      return
    } catch (err) {
      // ignore
    }
    try {
      await hardlinkPkg(to, opts)
      packageImportMethodLogger.debug({ method: 'hardlink' }) // ADDED
      auto = hardlinkPkg
      return
    } catch (err) {
      if (!err.message.startsWith('EXDEV: cross-device link not permitted')) throw err
      globalWarn(err.message)
      globalInfo('Falling back to copying packages from store')
      packageImportMethodLogger.debug({ method: 'copy' }) // ADDED
      auto = copyPkg
      await auto(to, opts)
    }
  }
}
```